packages/components/src/DateTimePickers/DateTime/Manager/Manager.component.js

```diff
@@ -1,197 +1,114 @@
-import React from 'react';
+/* eslint-disable react/no-unused-prop-types */
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import isSameSecond from 'date-fns/is_same_second';
 
 import { DateTimeContext } from '../Context';
 import {
-	checkSupportedDateFormat,
 	extractParts,
 	extractPartsFromDateAndTime,
-	extractPartsFromTextInput,
-} from '../date-extraction';
-import {
-	checkHours,
-	checkMinutes,
-	checkSeconds,
-	getTimeFormat,
-} from '../../Time/time-extraction';
-import {
-	HOUR_ERRORS,
-	MINUTES_ERRORS,
-	SECONDS_ERRORS,
-	FIELD_HOURS,
-	FIELD_MINUTES,
-	FIELD_SECONDS,
-	INPUT_ERRORS,
-} from '../constants';
+} from '../datetime-extraction';
+import { DATE_INPUT_ERRORS, TIME_INPUT_ERRORS } from '../constants';
 
-class ContextualManager extends React.Component {
-	static displayName = 'DateTime.Manager';
-	static propTypes = {
-		children: PropTypes.node,
-		dateFormat: PropTypes.string,
-		onChange: PropTypes.func,
-		required: PropTypes.bool,
-		selectedDateTime: PropTypes.oneOfType([
-			PropTypes.instanceOf(Date),
-			PropTypes.number,
-			PropTypes.string,
-		]),
-		useSeconds: PropTypes.bool,
-		useTime: PropTypes.bool,
-		useUTC: PropTypes.bool,
-	};
+function ContextualManager(props) {
+	// eslint-disable-next-line no-use-before-define
```

Comment:
`getDateOptions` function is declared later. eslint complains I used it before declaration, but it should not be a problem since functions declaration hoist to the top.

Reply:
Move the function declaration before